code/game/objects/effects/effect_system/effects_other.dm

```diff
@@ -66,11 +72,15 @@
 		processing = 0
 		var/turf/T = get_turf(holder)
 		if(T != oldposition)
-			if(!T.has_gravity())
+			if(!T.has_gravity() || !nograv_required)
 				var/obj/effect/particle_effect/ion_trails/I = PoolOrNew(effect_type, oldposition)
 				I.setDir(holder.dir)
-				flick("ion_fade", I)
-				I.icon_state = ""
+				if(istype(holder, /obj/item/device/flightpack))
```

Comment:
Don't do this, make a new proc

    /datum/effect_system/trail_follow/ion/proc/set_dir(/obj/effect/particle_effect/ion_trails/I)
         I.setDir(holder.dir)

Then override it for your new subtype

      /datum/effect_system/trail_follow/ion/proc/set_dir(/obj/effect/particle_effect/ion_trails/I)
           var/obj/item/device/flightpack/F = holder
           I.setDir(F.suit.user.dir)

Your way is less maintainable and harder to reuse for future work, ideally at the same time, take a look to see if the other trail_follow code can use the same procs, and if so, walk it up the type path to there.